services/api/database/migrations/20240114000000_environment_services.js

```diff
@@ -8,8 +8,11 @@ exports.up = async function(knex) {
         table.string('type', 300);
         table.timestamp('updated').notNullable().defaultTo(knex.fn.now());
         table.timestamp('created').notNullable().defaultTo(knex.fn.now());
-        table.unique(['name', 'environment'], {indexName: 'service_environment'});
     })
+    // create a unique index but drop any duplicates when the unique is added
+    // the deprecated setEnvironmentServices resolver will remove duplicates before adding them now too
+    .raw(`DELETE es1 FROM environment_service es1 INNER JOIN environment_service es2  WHERE  es1.id < es2.id AND es1.name = es2.name;`)
+    .raw(`ALTER TABLE environment_service ADD UNIQUE INDEX service_environment (name, environment);`)
```

Comment:
I'd convert this back to the native knex statement, just to avoid reliance on raw sql where possible?